oem/firstvoices/android/app/src/main/java/com/firstvoices/keyboards/FVShared.java

```
@@ -123,54 +129,90 @@ public static FVShared getInstance() {
+              if (keyboardObj == null) {
+                  KMLog.LogError(TAG, "keyboard Object in keyboards.json is null");
+                  continue;
+              }
+              JSONArray languageArray = keyboardObj.getJSONArray("languages");
+              if (languageArray == null) {
+                  KMLog.LogError(TAG, "languages Array in keyboards.json is null");
+                  continue;
+              }
+              String regionName = keyboardObj.getString("region");
+              FVRegion region = list.findRegion(regionName);
+              if(region == null) {
+                  region = new FVRegion(regionName);
+                  list.add(region);
+              }
+
+              String kbID = keyboardObj.getString("id");
+              String kbName = keyboardObj.getString("name");
+              JSONObject languageObj = languageArray.getJSONObject(0);
+              String lgID = languageObj.getString("id").toLowerCase(); // Normalize language Id
+              String lgName = languageObj.getString("name");
+
+              // Override European keyboard info
+              if (kbID.equalsIgnoreCase("sil_euro_latin")) {
+                kbName = "English";
+                lgID = "en";
+                lgName = "English";
+              } else if (kbID.equalsIgnoreCase("basic_kbdcan")) {
+                kbName = "Français";
+                lgID = "fr-ca";
+                lgName = "French";
+              }
+
+              FVKeyboard kbd = new FVKeyboard(
+                  kbID,
+                  kbName,
+                  kbID, // unused legacy ID
+                  keyboardObj.getString("version"),
+                  lgID,
+                  lgName);
+
+              region.keyboards.add(kbd);
+          }
+      } catch (Exception e) {
+          String errorMsg = "loadRegionList had malformed keyboard list";
+          KMLog.LogException(TAG, errorMsg, e);
+          // Crash the app
+          throw new Error(errorMsg);
```

This is an antipattern -- masking the original error in the thrown exception. Should re-throw e.